central/group/datastore/datastore_impl.go

```diff
@@ -185,45 +179,33 @@ func (ds *dataStoreImpl) Mutate(ctx context.Context, remove, update, add []*stor
 	return nil
 }
 
-func (ds *dataStoreImpl) Remove(ctx context.Context, props *storage.GroupProperties) error {
+func (ds *dataStoreImpl) Remove(ctx context.Context, force bool, props *storage.GroupProperties) error {
 	if ok, err := groupSAC.WriteAllowed(ctx); err != nil {
 		return err
 	} else if !ok {
 		return sac.ErrResourceAccessDenied
 	}
 
-	if err := ValidateProps(props); err != nil {
-		return errox.InvalidArgs.CausedBy(err)
-	}
-
 	// Lock to ensure synchronization between the get and delete
 	ds.lock.Lock()
 	defer ds.lock.Unlock()
 
-	propsID := props.GetId()
-
-	// TODO(ROX-11592): Once the deprecation of retrieving groups by their properties is fully deprecated, this condition
-	// can be removed and groups shall only be retrievable via their id.
-	if propsID == "" {
-		id, err := ds.findGroupIDByProps(ctx, props)
-		if err != nil {
-			return err
-		}
-		// Use the id of the retrieved group to delete
-		propsID = id
+	groupID, err := ds.validateAndPrepGroupForDeleteNoLock(ctx, props, force)
+	if err != nil {
+		return err
 	}
 
-	return ds.storage.Delete(ctx, propsID)
+	return ds.storage.Delete(ctx, groupID)
 }
 
-func (ds *dataStoreImpl) RemoveAllWithAuthProviderID(ctx context.Context, authProviderID string) error {
+func (ds *dataStoreImpl) RemoveAllWithAuthProviderID(ctx context.Context, force bool, id string) error {
```

Comment:
Why the renaming `authProviderID => id`? I found `authProviderID` helpful as it's more explicit and `id` in the group datastore setting sounds like group ID